### ISSUE TITLE
Mouse/Touch input with scaled BackBuffer

### ISSUE THREAD

**Issue by nkast** [MonoGame/MonoGame#5570]
Touch input is incorrect when used with a scaled back buffer.

The following code will offset the mouse/touch by four times.
```
graphics.PreferredBackBufferWidth /= 4;
            graphics.PreferredBackBufferHeight /= 4;
            this.Window.ClientSizeChanged += (s, e) =>
            {
                graphics.PreferredBackBufferWidth /= 4;
                graphics.PreferredBackBufferHeight /= 4;
                graphics.ApplyChanges();
            };
```

So far Tested on W8.1 & WP8.1.

The string "Mouse" is supposed to be drawn below the mouse cursor.
(Mouse Cursor not shown, it's on the upper-left corner of F-rame))
![touchwithscaledbackbuffer](https://cloud.githubusercontent.com/assets/3018589/23781860/ca7efdb8-0558-11e7-89da-a2deddd5d433.png)

![touchwithscaledbackbufferwp](https://cloud.githubusercontent.com/assets/3018589/23781863/cce39f14-0558-11e7-8f7f-ee185e7209cd.png)

Here are some related debug values
```
this.Window.ClientBounds
0  0  500  768
this.GraphicsDevice.PresentationParameters.BackBufferWidth
125
this.GraphicsDevice.PresentationParameters.BackBufferHeight
192
TouchPanel.DisplayWidth
125
TouchPanel.DisplayHeight
192
```

**Comment by tomspilman** [comment]
> The following code will offset the mouse/touch by four times.

I think you'll find that this is the same behavior on WP7 and XNA.

You had to manually change `TouchPanel.DisplayWidth` and `TouchPanel.DisplayHeight` to match your back buffer size to make the touch inputs scale.  It did not do this automatically for you.  We can probably find some docs somewhere to verify that.

**Comment by nkast** [comment]
> I think you'll find that this is the same behavior on WP7 and XNA.

Really? Oh, ok. I will close it then.

**Comment by tomspilman** [comment]
> Really? Oh, ok. I will close it then.

Someone should double check me...  but that is what I remember.

**Comment by nkast** [comment]
I kind of remember it was that way after you said it.
It's kind of impossible to test WP7 those days on Win10, but I could test mouse on a fullscreen XNA within the next days.

**Comment by tomspilman** [comment]
> It's kind of impossible to test WP7 those days

Yeah I just think that the only way it worked with touch and fake mouse from touch was on WP7.

On PC the touch screen stuff didn't do anything at all.

So docs might be the only way to verify that.

**Comment by nkast** [comment]
I tested this with XNA and on a couple MG platforms.

- XNA scales mouse input to the backbuffer dimension.
- W8.1 & WP8.1, now behave similar to XNA. Fixed by #5520, Listening to `CoreIndependentInputSource` fixed the original issue.
- W10 also fixed by #5520.
- Desktop GL scales mouse input to the backbuffer dimension similar to XNA.
- Desktop DX has the bug I originally posted for W8.1, mouse input is the position from Window origin, not scaled to the backbuffer.

Test projects: https://1drv.ms/u/s!AgIeh29TX5UhkPUa1VKlK7N4X0KM3Q

Besides that, there are issues on every platform.

- For one, no platform ends up with the same backbuffer size. I have the same resize code on every project.
```
graphics.PreferredBackBufferWidth /= 2;
            graphics.PreferredBackBufferHeight /= 2;
            this.Window.ClientSizeChanged += (s, e) =>
            {
                graphics.PreferredBackBufferWidth /= 2;
                graphics.PreferredBackBufferHeight /= 2;
                graphics.ApplyChanges();
            };
```
- GL fires ClientSizeChanged with PreferredBackBufferWidth = 0 during initialization.
- W10 doesn't scale the backbuffer to fill the window.

**Comment by nkast** [comment]
I am starting to thinking that XNA didn't do anything special with Mouse and the Backbuffer resolution.
In fullscreen the desktop resolution was changed to match the preferred resolution.
Similarly In window mode the window size was changed to match the preferred resolution.
The window size was always the same as the backbuffer. The Mouse class didn't had knowledge of the Game class or the GaphicsDevice, it reported the mouse state relative to the assigned WindowHandle.

IMO it's important to keep the behavior the same across all platforms. We can't have one platform reporting mouse position relative to the back-buffer while others require  the user to scale it from (BackBufferSize/windowSize). Perhaps the right solution is to introduce a `.DisplayWidth`/`DisplayHeight` property.
